src/main/anovos/data_ingest/data_ingest.py

```diff
@@ -41,13 +57,188 @@ def read_dataset(spark, file_path, file_type, file_configs={}):
+        Otherwise its DataType will be treated as String. (Default value = 0.5)
 
     Returns
     -------
     DataFrame
 
     """
     odf = spark.read.format(file_type).options(**file_configs).load(file_path)
+    if not isinstance(parquet_conversion, bool):
+        raise TypeError("parquet_conversion must be boolean")
+    if parquet_conversion:
+        if not parquet_output_path:
+            raise ValueError(
+                "parquet_output_path cannot be blank for parquet conversion"
+            )
+    if not isinstance(treatment, bool):
+        raise TypeError("treatment must be boolean")
+    if treatment:
+        if id_cols:
+            for col in id_cols:
+                if col not in odf.columns:
+                    raise ValueError("Column not in the input dataframe: " + col)
+        if not isinstance(threshold_num, int):
+            raise TypeError("threshold_num must be integer")
+        if not isinstance(threshold_ratio, float):
+            raise TypeError("threshold_ratio must be float")
+        if threshold_ratio < 0 or threshold_ratio > 1:
+            raise ValueError("threshold_ratio value must be between 0 and 1")
+        if not isinstance(threshold_string, float):
+            raise TypeError("threshold_string must be float")
+        if threshold_string < 0 or threshold_string > 1:
+            raise ValueError("threshold_string value must be between 0 and 1")
+        if not isinstance(persist_use, bool):
+            raise TypeError("persist_use must be boolean")
+        if persist_use:
+            if persist_option == "DISK_ONLY":
+                odf = odf.persist(pyspark.StorageLevel.DISK_ONLY)
+            elif persist_option == "DISK_ONLY_2":
+                odf = odf.persist(pyspark.StorageLevel.DISK_ONLY_2)
+            elif persist_option == "MEMORY_AND_DISK":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_AND_DISK)
+            elif persist_option == "MEMORY_AND_DISK_2":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_AND_DISK_2)
+            elif persist_option == "MEMORY_AND_DISK_SER":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_AND_DISK_SER)
+            elif persist_option == "MEMORY_AND_DISK_SER_2":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_AND_DISK_SER_2)
+            elif persist_option == "MEMORY_ONLY":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_ONLY_2)
+            elif persist_option == "MEMORY_ONLY_SER":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_ONLY_SER)
+            elif persist_option == "MEMORY_ONLY_SER_2":
+                odf = odf.persist(pyspark.StorageLevel.MEMORY_ONLY_SER_2)
+            elif persist_option == "OFF_HEAP":
+                odf = odf.persist(pyspark.StorageLevel.OFF_HEAP)
+            else:
+                raise TypeError(
+                    "Input persist_option does not exist in Spark StorageLevel"
+                )
```

Comment:
Why do you have the user pass a string instead of the actual storage level? You could just pass `persist_option=pyspark.StorageLevel.MEMORY_AND_DISK` instead of translating between an error-prone string and the actual StorageLevel.

Reply:
Resolved with https://github.com/anovos/anovos/pull/268/commits/dd553da3d2fc2ba0207a6e08d993a481743525ca